### PR TITLE
Enable ParquetFileFomat splitable  capability except build index

### PR DESCRIPTION
**Pull request by LuciferYang**
## What changes were proposed in this pull request?

Add a splitable status and forbidSplitable method to ParquetFileFomat, then ParquetFileFomat.isSplitable should return false when "create Index" and "refresh Index", other case should return true to increasing task parallelism and block local read ratio, then we can keep same behavior as Spark.

## How was this patch tested?

existing test

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala [hunk 1]
```diff
@@ -271,9 +275,7 @@ class ParquetFileFormat
   override def isSplitable(
       sparkSession: SparkSession,
       options: Map[String, String],
-      path: Path): Boolean = {
-      !sparkSession.conf.get(SQLConf.OAP_PARQUET_ENABLED)
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala]: If you delete this, this config will be no longer used?

File: src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapSuite.scala [hunk 2]
```diff
@@ -74,11 +74,13 @@ class OapSuite extends QueryTest with SharedOapContext with BeforeAndAfter {
...
     val defaultMaxBytes = sqlConf.getConf(SQLConf.FILES_MAX_PARTITION_BYTES)
-    sqlConf.setConf(SQLConf.FILES_MAX_PARTITION_BYTES, 100L)
+    val maxPartitionBytes = 100L
+    sqlConf.setConf(SQLConf.FILES_MAX_PARTITION_BYTES, maxPartitionBytes)
     val numTasks = sql("select * from parquet_table").queryExecution.toRdd.partitions.length
     try {
       sql("create oindex parquet_idx on parquet_table (a)")
-      assert(numTasks == parquetPath.listFiles().count(_.getName.endsWith(".index")))
+      assert(numTasks == parquetPath.listFiles().filter(_.getName.endsWith(".parquet"))
+        .map(f => Math.ceil(f.length().toDouble / maxPartitionBytes).toInt).sum)
```
Comment (adrian-wang) [inline src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapSuite.scala]: also test if we can forbid it successfully?

File: src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala [hunk 3]
```diff
@@ -271,9 +275,7 @@ class ParquetFileFormat
...
-  }
+      path: Path): Boolean = splitable
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala]: only when oap parquet enabled?

Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala]: ParquetFileFormat.isSplitable should return true except for create or refresh index cmd and splitable default value is true.

File: src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapSuite.scala (hunk 2 above)
Comment (LuciferYang) [reply src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapSuite.scala]: Let me think about how to assert  forbid  operation is successful

Comment (LuciferYang) [reply src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapSuite.scala]: I didn't think of a good way. Do you have any suggestions? @adrian-wang

Comment (LuciferYang) [reply src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapSuite.scala]: I add a new case named "forbidSplit method in Parquet File Format" to test effect of forbidSplit mehtod, Is it enough? @adrian-wang

**Comment by adrian-wang** [comment]
merged to master and branch-0.3

**Pull request merged**